warehouse/integrations/bigquery/bigquery.go

```diff
@@ -793,41 +915,36 @@ func (bq *BigQuery) dropDanglingStagingTables(ctx context.Context) bool {
 	return delSuccess
 }
 
-func (bq *BigQuery) IsEmpty(ctx context.Context, warehouse model.Warehouse) (empty bool, err error) {
-	empty = true
+func (bq *BigQuery) IsEmpty(ctx context.Context, warehouse model.Warehouse) (bool, error) {
 	bq.warehouse = warehouse
 	bq.namespace = warehouse.Namespace
 	bq.projectID = strings.TrimSpace(warehouseutils.GetConfigValue(project, bq.warehouse))
-	bq.logger.Infof("BQ: Connecting to BigQuery in project: %s", bq.projectID)
+
+	var err error
 	bq.db, err = bq.connect(ctx, BQCredentials{
 		ProjectID:   bq.projectID,
 		Credentials: warehouseutils.GetConfigValue(credentials, bq.warehouse),
 	})
 	if err != nil {
-		return
+		return false, fmt.Errorf("connecting to bigquery: %v", err)
 	}
 	defer func() { _ = bq.db.Close() }()
 
 	tables := []string{"tracks", "pages", "screens", "identifies", "aliases"}
 	for _, tableName := range tables {
-		var exists bool
-		exists, err = bq.tableExists(ctx, tableName)
-		if err != nil {
-			return
-		}
-		if !exists {
+		if exists, err := bq.tableExists(ctx, tableName); err != nil {
+			return false, fmt.Errorf("checking if table %s exists: %v", tableName, err)
+		} else if !exists {
 			continue
 		}
-		count, err := bq.GetTotalCountInTable(ctx, tableName)
+
+		metadata, err := bq.db.Dataset(bq.namespace).Table(tableName).Metadata(ctx)
```

Comment:
This line doesn't seem to have coverage (many others don't but I think it would make sense to have it here). wdyt?